modules/terraform/count.go

```diff
@@ -0,0 +1,90 @@
+package terraform
+
+import (
+	"errors"
+	"regexp"
+	"strconv"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+// ResourceCount represents counts of resources affected by terraform apply/plan/destroy command.
+type ResourceCount struct {
+	Add     int
+	Change  int
+	Destroy int
+}
+
+// Regular expressions for terraform commands stdout pattern matching.
+const (
+	applyRegexp             = `Apply complete! Resources: (\d+) added, (\d+) changed, (\d+) destroyed\.`
+	destroyRegexp           = `Destroy complete! Resources: (\d+) destroyed\.`
+	planWithChangesRegexp   = `(\033\[1m)?Plan:(\033\[0m)? (\d+) to add, (\d+) to change, (\d+) to destroy\.`
+	planWithNoChangesRegexp = `No changes\. Infrastructure is up-to-date\.`
+)
+
+const getResourceCountErrMessage = "Can't parse Terraform output"
+
+// GetResourceCount parses stdout/stderr of apply/plan/destroy commands and returns number of affected resources.
+// This will fail the test if given stdout/stderr isn't a valid output of apply/plan/destroy.
+func GetResourceCount(t *testing.T, stdout string) *ResourceCount {
+	cnt, err := GetResourceCountE(t, stdout)
+	require.NoError(t, err)
+	return cnt
+}
+
+// GetResourceCountE parses stdout/stderr of apply/plan/destroy commands and returns number of affected resources.
+func GetResourceCountE(t *testing.T, stdout string) (*ResourceCount, error) {
+	var pattern *regexp.Regexp
+	var err error
+	var matches []string
+	var add, change, destroy int
```

Comment:
Is there a reason you are declaring these at the top?

Reply:
rewritten in a more go-idiomatic way